### PR TITLE
Bulk api support + some other fixes/improvements

### PR DESCRIPTION
**Pull request by chrisbenincasa**

### CODE REVIEW

File: lib/logstash/outputs/loggly.rb [hunk 1]
```diff
@@ -91,31 +111,58 @@ class LogStash::Outputs::Loggly < LogStash::Outputs::Base
...
   end # def receive
 
   public
   def format_message(event)
+    @logger.debug("Actual event: #{event.to_json}")
     event.to_json
   end
 
+  def flush(events, close=false)
+    # Avoid creating a new string for newline every time
```
Comment (devshorts) [inline lib/logstash/outputs/loggly.rb:152]: Lets comment this function indicating this ist he loggly bulk api

**Review by devshorts — APPROVED** [review]
Looks sane enough to me. Is the Stud::Buffer class the recommended approach for ruby buffering?

**Comment by chrisbenincasa** [comment]
@devshorts I saw it used in a few other logstash plugins I poked around in, but don't know enough about Ruby to really know 😆

**Comment by richid** [comment]
Looks good enough to me but I'm no expert here. I'd want to run this for longer periods of time to ensure there are no resource leaks or other issues.

**Comment by devshorts** [comment]
Def agree with @richid

**Comment by chrisbenincasa** [comment]
Most definitely!

**Pull request merged**